src/server/hll_family_test.cc

```diff
@@ -194,6 +194,13 @@ TEST_F(HllFamilyTest, MergeOverlapping) {
 }
 
 TEST_F(HllFamilyTest, MergeInvalid) {
+  const bool kIsClusterModeEnabled = IsClusterEnabledOrEmulated();
+  // We have a bug here unrelated to cluster. Basically we trigger the check fail
```

Comment:
That was a bug before if the keys belonged to different shards

Reply:
So we have a known issue with hll that we have not fixed?

Reply:
Two issues! multikey variant of this https://github.com/dragonflydb/dragonfly/pull/4994#discussion_r2060533271 also fails. See comment. 

The reason we discovered this pr changed the hashing in cluster mode. Some tests enforced single shard operations and with this pr they became multi shard uncovering those corner cases.

I will open issues about it.

p.s. I tried to preserve test semantics where needed. For some tests the semantic change (from single to multi shard) was not really important so I adjusted the assertions instead.